sql/sys_vars.cc

```diff
@@ -633,6 +633,22 @@ static Sys_var_charptr Sys_character_sets_dir(
        READ_ONLY GLOBAL_VAR(charsets_dir), CMD_LINE(REQUIRED_ARG),
        IN_FS_CHARSET, DEFAULT(0));
 
+static bool check_unsupported_engine(sys_var *self, THD *thd, set_var *var)
+{
+  LEX_CSTRING name= var->value->name;
+  plugin_ref plugin;
+  handlerton *hton;
+  plugin= ha_resolve_by_name(0, &name, true);
+  hton= plugin_hton(plugin);
+  if (ha_check_storage_engine_flag(hton, HTON_TEMPORARY_NOT_SUPPORTED))
+  {
+    sql_print_error("Unknown/unsupported storage engine: %s", hton_name(hton)->str);
```

Comment:
the engine cannot be unknown, `Sys_var_plugin` will guarantee that the engine is installed and active.

I'd just use
```c
my_error(ER_ILLEGAL_HA_CREATE_OPTION, MYF(0), hton_name(hton)->str, "TEMPORARY");
```